rfcs/rfc-cloud-events-for-name-request.md

```diff
@@ -0,0 +1,272 @@
+be used within the context of the development teams at BC Registries.  As such, this section will not contain an 
+exhaustive description of CloudEvent attributes.  For more details of CloudEvent attributes, please refer to the 
+CloudEvents specification as provided in the references section of this document.
+
+
+| Attribute Name  | Attribute Type | Data Type      | Example Value                        | Required | Notes      |
+|-----------------|----------------|----------------|--------------------------------------|----------|------------|
+| specversion     | context        | string         | 1.0.1                                | Y        |            |
+| source          | context        | URI-reference  | /sbc_pay_api/invoice/<INVOICE_ID>    | Y        |            |
+| id              | context        | string         | 16fd2706-8baf-433b-82eb-8c7fada847da | Y        | use uuidv4 |
+| type            | context        | string         | bc.registry.nr_payment.complete      | Y        |            |
+| datacontenttype | context        | string         | application/json                     | N        |            |
+| time            | context        | timestamp      |                                      | N        |            |
+| identifier      | custom         | string         | 111111(FILING_ID)                    | N        |            |
+| data            | data           | no restriction | datacontenttype format               | N        |            |
+
+Of particular interest are the following items:
+
+**Idempotent use of a message (`source` + `id`)** - to ensure that producers of messages are publishing distinct events, 
+a unique combination of `source` + `id` must be used.  The subscribers will have the responsibility of using the 
+`source` + `id` combination to determine how to process messages in an idempotent manner if that is required for a given
+subscriber's use case.  This will likely mean that the `source` + `id` will need to be recorded/tracked in order to 
+determine if a message has already been processed.
+
+**source** - identifies the context in which an event happened. 
+
+**id** - identifies the event.  Use uuidv4 for the id which is provided by the uuid python module. i.e. `uuid.uuid4()`  
+
+**data** - contains information specific to the event.  Typically this will contain a json payload and services consuming
+this data payload will do most of the core processing of an event with the information provided in this payload.  The content
+type of this field is encoded in the media format specified by the `datacontenttype` attribute.
+
+
+# Detailed design
+
+The diagram below provides the flows that will need to be updated to support our CloudEvent implementation.  Specifically,
+the integration points are the areas where work will need to be done.
+
+![Integration Points Diagram](rfc-cloud-events-for-name-request/cloud_events_diagram.png)
```

Comment:
May need some changes to the diagram.
Auth and Pay won't need any subscriber when an NR is PAID, as no records has to be changed when payment is received as the logic to handle what happens when a payment is received is fully up to the integrated applications (lear, ppr, nr etc.).  
The events which are of interest for auth and pay would be any state change and identifier changes.
- When a state change occurs on NR and needs to be affiliated to an account.
- When a state change occurs on NR and the identifier changes from temporary identifier to final identifier. (NRL -> NR).

Also can the type be decided which would be more helpful on the subscriber side to not do lot of logic based on the statuses?